src/components/PrivateRoute.tsx

```diff
@@ -0,0 +1,24 @@
+import { JSXElementConstructor } from "react";
```

Comment:
Just pulling this out as its own component, and setting up types to accept functional or class components